src/test/java/com/epam/tc/hw2/ex1/FirstExerciseTest.java

```diff
@@ -0,0 +1,157 @@
+        List<WebElement> images = driver.findElements(By.cssSelector(imagesCssLocators));
+        assertThat(images.size()).isEqualTo(imageCssLocatorToTextAfterImage.size());
+
+        for (String imageClassName : imageCssLocatorToTextAfterImage.keySet()) {
+            WebElement image = driver.findElement(By.cssSelector(imageClassName));
+            assertThat(isImageDisplayedAfterWait(image)).isEqualTo(true);
+            WebElement textElement = getTextNodeFromImageNode(image);
+            assertThat(textElement.getText()).isEqualTo(
+                imageCssLocatorToTextAfterImage.get(imageClassName)
+            );
+        }
+    }
+
+
+    // 8. Assert that there is the iframe with “Frame Button” exist;
+    // 9. Switch to the iframe and check that there is “Frame Button” in the iframe;
+    // there's noway to check if frame w/ button exists but to
+    // switch to every frame and check if there's a button in some frame
+    // so making sure 8 is true necessarily implies that actions in 9 are already done
+    public void checkFrameWithButtonExists(String frameButtonValue) {
+        List<WebElement> frames = driver.findElements(By.tagName("iframe"));
+        boolean frameWithButtonExists = false;
+        for (WebElement frame : frames) {
+            wait.until(ExpectedConditions.frameToBeAvailableAndSwitchToIt(frame));
+            List<WebElement> frameButtons = driver.findElements(By.xpath(
+                String.format("//*[@value='%s']", frameButtonValue))
+            );
+            int numButtons = frameButtons.size();
+            switchToOriginalWindow();
+            if (numButtons > 0) {
+                frameWithButtonExists = true;
+                break;
+            }
+        }
+        assertThat(frameWithButtonExists).isEqualTo(true);
+    }
+
+    // 11. Assert that there are 5 items in the Left Section
+    // are displayed and they have proper text;
+    public void checkLeftSectionHasItemsWithProperText(List<String> expectedLeftSectionItemsText) {
+        // go directly to tag w/ text
+        List<WebElement> leftSectionElements = driver.findElements(
+            By.cssSelector(".sidebar-menu.left > * > * > span")
+        );
+        assertThat(leftSectionElements.size()).isEqualTo(expectedLeftSectionItemsText.size());
+        List<String> leftSectionElementsTexts = leftSectionElements
+            .stream()
+            .map(WebElement::getText)
+            .sorted()
+            .collect(Collectors.toList());
+        assertThat(leftSectionElementsTexts).isEqualTo(expectedLeftSectionItemsText);
+    }
+
+    // 10. Switch to original window back;
+    // used in checkFrameWithButtonExists multiple times
+    // to switch from frame to original window
+    public void switchToOriginalWindow() {
+        driver.switchTo().defaultContent();
+    }
+
+    public WebElement getTextNodeFromImageNode(WebElement image) {
+        WebElement parent = image.findElement(By.xpath("./../.."));
```

Comment:
Such locator hard understandable in future. Only you know what does it mean. Can it be changed on clear one?